Classes/Handler/FileUploadHandler.php

```diff
@@ -49,27 +84,28 @@ public function uploadFile(InterestRequestInterface $request): ResponseInterface
             );
         }
 
-        $configuration = $this->objectManager->getConfigurationProvider()->getSettings();
         $httpClient = $this->objectManager->get(Client::class);
         $fileBaseName = basename($data['data']['url']);
-        $storagePath = $configuration['persistence']['fileUploadFolderPath'];
-        [$storageId, $subFolderPath] = explode(':', $storagePath);
-        $storage = $this->objectManager->get(StorageRepository::class)->findByUid((int)$storageId);
 
         if ($data['data']['fileData']) {
             $fileBaseName = $data['data']['name'];
-            $stream = fopen('fileadmin/' . $fileBaseName, 'w');
-            stream_filter_append($stream, 'convert.base64-decode', STREAM_FILTER_WRITE);
-            fwrite($stream, $data['data']['fileData']);
-            fclose($stream);
+            $downloadFolder->createFile($fileBaseName);
+
+            if ($downloadFolder->hasFile($fileBaseName)) {
```

Comment:
Maybe wise with an overwrite flag in the request? I'm not sure you'll always want to overwrite the data. Something along the lines of:
1. Fail
2. Create with a different name (add number to end of file)
3. Replace content
4. Use the existing file record unchanged